sci-electronics/systemc: Clean up.

omasanori opened the pull request:
- Stop to inherit unused eutils.eclass.
- Remove CXX=$(tc-getCXX) from econf.

mgorny commented on sci-electronics/systemc/systemc-2.3.0-r1.ebuild:
```diff
@@ -3,7 +3,7 @@
 EAPI=4
 
-inherit eutils multilib toolchain-funcs autotools-utils
+inherit multilib toolchain-funcs autotools-utils
```
it is not unused. The ebuild uses `epatch` which used to reside in `eutils`. The correct solution would be to replace it with `epatch` eclass.

omasanori replied:
OK, I will fix them.

mgorny commented on sci-electronics/systemc/systemc-2.3.0-r1.ebuild:
```diff
@@ -45,7 +45,8 @@ src_prepare() {
 }
 
 src_configure() {
-	econf $(use_enable static-libs static) CXX=$(tc-getCXX)
```
Why are you removing it? Are you sure that this does not change behavior? Note that some configure files divert from the default GNU behavior, in which case we *want* to force CXX. For example, some projects default to using clang when available.

omasanori replied:
I removed them because they use ${CHOST}-g++ regardless of the existence of `CXX=$(tc-getCXX)`, at least at 2.3.0 and 2.3.1. However, indeed, there must be any reason why the argument was added. I will restore them for now and check again later.

omasanori commented:
Updated. Thank you for reviewing, @mgorny !
By the way I added TODO comments to `CXX=$(tc-getCXX)` lines because the reason of them is unclear to me unlike `sed` lines which are already commented by the original authors. Is it acceptable? If not, I will remove them.

SoapGentoo commented:
@omasanori you could possibly patch the whole configure.ac and build system not to require the CXX override?

omasanori commented:
@SoapGentoo Indeed but it needs more investigation. Is it better to close this PR for now and make complete fix later instead of step-by-step PRs?

mgorny commented:
@omasanori, yes, having split commits will be better. FWICS, the only meaningful change here is replacing the eutils inherit, so feel free to submit a commit doing exactly that.

mgorny commented:
Ping.

omasanori commented:
I am sorry for the delay. I will make revised PR for `eutils`-to-`epatch` migration.

mgorny commented:
Closing due to no activity. Feel free to reopen when you have time to work on it again (or create a new PR). Please note that GH will not permit you to reopen if you push new commits first.
